app/code/Magento/Vault/Model/AbstractPaymentTokenFactory.php

```diff
@@ -35,9 +50,11 @@ public function __construct(ObjectManagerInterface $objectManager)
      */
     public function create()
     {
-        /** @var PaymentTokenInterface $paymentToken */
-        $paymentToken = $this->objectManager->create(PaymentTokenInterface::class);
-        $paymentToken->setType($this->getType());
-        return $paymentToken;
+        return $this->paymentTokenFactory->create($this->getType());
     }
+
+    /**
+     * @return string
+     */
+    abstract public function getType();
```

Comment:
no need to specify abstract public getType method here
we get it from PaymentTokenInterfaceFactory 
```php
interface PaymentTokenInterfaceFactory
{
    /**
     * Create payment token entity
     * @return PaymentTokenInterface
     */
    public function create();

    /**
     * Return type of payment token
     * @return string
     */
    public function getType();
}

```